pkg/northbound/gnmi/v2/set.go

```diff
@@ -211,35 +217,52 @@ func (s *Server) Set(ctx context.Context, req *gnmi.SetRequest) (*gnmi.SetRespon
 	return nil, ctx.Err()
 }
 
-func (s *Server) getTargetInfo(ctx context.Context, targets map[configapi.TargetID]*targetInfo, idPrefix string, id configapi.TargetID) (*targetInfo, error) {
+func (s *Server) getTargetInfo(ctx context.Context, targets map[configapi.TargetID]*targetInfo,
+	overrides *configapi.TargetVersionOverrides, idPrefix string, id configapi.TargetID) (*targetInfo, error) {
 	targetID := configapi.TargetID(idPrefix)
 	if len(id) > 0 {
 		targetID = id
 	}
 
+	// Just return the target info if we already have it
 	if target, found := targets[targetID]; found {
 		return target, nil
 	}
 
-	configurable, err := s.getTargetConfigurable(ctx, topoapi.ID(targetID))
-	if err != nil {
-		log.Warn(err)
-		return nil, err
+	// Use the type/version overrides if they are specified
+	var targetType configapi.TargetType
+	var targetVersion configapi.TargetVersion
+	persistent := true // TODO determine whether this is an appropriate default
+	if ttv, ok := overrides.Overrides[string(targetID)]; ok {
+		targetType = ttv.TargetType
+		targetVersion = ttv.TargetVersion
+	} else {
+		// Otherwise, extract the information from the Configurable aspect of the target topo entity
+		configurable, err := s.getTargetConfigurable(ctx, topoapi.ID(targetID))
+		if err != nil {
+			log.Warn(err)
+			return nil, err
+		}
+		targetType = configapi.TargetType(configurable.Type)
+		targetVersion = configapi.TargetVersion(configurable.Version)
+		persistent = configurable.Persistent
 	}
 
-	modelPlugin, ok := s.pluginRegistry.GetPlugin(configapi.TargetType(configurable.Type), configapi.TargetVersion(configurable.Version))
+	// Find the model plugin using the target type and version
+	modelPlugin, ok := s.pluginRegistry.GetPlugin(targetType, targetVersion)
 	if !ok {
-		err = errors.NewNotFound("model %s (v%s) plugin not found", configurable.Type, configurable.Version)
+		err := errors.NewNotFound("model %s (v%s) plugin not found", targetType, targetVersion)
 		log.Warn(err)
 		return nil, err
 	}
```

Comment:
It looks TargetTypeVersion map in newTransaction function is not initialized with these info. Since you are here you can make the change to pass it down.